puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-x-develop-release.groovy

```diff
@@ -10,18 +10,23 @@ pipeline {
         ansiColor('xterm')
     }
 
-    environment {
-        ruby_version = "2.5"
-    }
-
     stages {
         stage('Copy Source') {
             steps {
                 script {
-                    sourcefile_paths = pwd(tmp: true)
+                    artifact_path = "${pwd()}/artifacts"
                     source_project_name = "${project_name}-${git_ref}-source-release"
-                    copyArtifacts(projectName: source_project_name, target: sourcefile_paths, selector: {lastSuccessful: true})
+                    copyArtifacts(projectName: source_project_name, target: artifact_path)
+                    sourcefile_paths = list_files("${artifact_path}/*").findAll { it.endsWith('.gem') || it.endsWith('tar.bz2') || it.endsWith('tar.gz') }
+                }
+            }
+        }
+        stage('Setup Environment') {
```

Comment:
This env is only needed for the RPM so I think this should be part of `Build RPM`

Reply:
Fair, Deb does not need the copy artifacts either so I have updated this to further encapsulate the two paths.

Reply:
Example run:https://ci.theforeman.org/blue/organizations/jenkins/hammer-cli-katello-master-package-release/detail/hammer-cli-katello-master-package-release/66/pipeline

Reply:
The deb side does need the variable for the job (`source_project_name`). Perhaps that should be in `environment {}`?